### PR TITLE
Rename dhcp related interface options to dhcp-* and allow interface specific config file

### PR DESCRIPTION
**Pull request by BarbarossaTM**

### CODE REVIEW

File: executor-scripts/linux/dhcp [hunk 1]
```diff
@@ -14,20 +14,29 @@ determine_implementation() {
...
-		[ -n "$IF_HOSTNAME" ] && optargs="$optargs -h $IF_HOSTNAME"
-		[ -n "$IF_VENDOR" ] && optargs="$optargs -i $IF_VENDOR"
-		[ -n "$IF_CLIENT" ] && optargs="$optargs -i $IF_CLIENT"
-		[ -n "$IF_LEASETIME" ] && optargs="$optargs -l $IF_LEASETIME"
+		[ -n "$IF_DHCP_HOSTNAME" ] && optargs="$optargs -h $IF_DHCP_HOSTNAME"
+		[ -n "$IF_DHCP_VENDOR" ] && optargs="$optargs -i $IF_DHCP_VENDOR"
+		[ -n "$IF_DHCP_CLIENT_ID" ] && optargs="$optargs -i $IF_DHCP_CLIENT_ID"
+		[ -n "$IF_DHCP_LEASETIME" ] && optargs="$optargs -l $IF_DHCP_LEASETIME"
+
+		# Specific config file given?
```
Comment (kaniini) [inline executor-scripts/linux/dhcp:22]: As mentioned in IRC, this section should be placed under `dhclient)` part.

Comment (BarbarossaTM) [reply executor-scripts/linux/dhcp:24]: Fixed

**Comment by kaniini** [comment]
It looks fine to me, but I wonder if perhaps we should just namespace *all* of the addons at once.

**Comment by BarbarossaTM** [comment]
As far as I can see DHCP is the last thing which is a good fit for it's own namespace and were are good on the rest.

**Comment by kaniini** [comment]
One thing missing here is a lack of doc updates for it.  I am pretty sure interfaces.5 discusses the DHCP options (and now that they are properly namespaced, perhaps moving them into interfaces-dhcp.5 makes sense?)

**Pull request merged**